iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp

```diff
@@ -595,6 +595,95 @@ class TransposeReshapeGenericDotGeneral
   }
 };
 
+class ScatterRank0Value : public OpRewritePattern<mhlo::ScatterOp> {
+ public:
+  using OpRewritePattern<mhlo::ScatterOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mhlo::ScatterOp op,
+                                PatternRewriter &rewriter) const override {
+    auto operand = op.operand();
+    auto indices = op.scatter_indices();
+    auto updates = op.updates();
+
+    auto operandTy = operand.getType().dyn_cast<RankedTensorType>();
+    auto indicesTy = indices.getType().dyn_cast<RankedTensorType>();
+    auto updatesTy = updates.getType().dyn_cast<RankedTensorType>();
+    if (!operandTy || !indicesTy || !updatesTy) return failure();
+
+    if (indicesTy.getRank() != 1 || !indicesTy.hasStaticShape() ||
+        updatesTy.getRank() != 0)
+      return failure();
+
+    auto dimNumbers = op.scatter_dimension_numbers();
+
+    // We only have one dim for shape so this should be 0.
+    if (dimNumbers.getIndexVectorDim() != 0) return failure();
+
+    // Require canonicalize scatter order.
+    // TODO(suderman): Transpose to canonicalized order.
+    for (auto en : llvm::enumerate(dimNumbers.getScatterDimsToOperandDims())) {
+      if (en.index() != en.value()) return failure();
+    }
+
+    // Inserted window dims should be in order. Technically we just need to
+    // check they are all contained.
+    for (auto en : llvm::enumerate(dimNumbers.getInsertedWindowDims())) {
+      if (en.index() != en.value()) return failure();
+    }
+
+    // This should be empty
+    if (dimNumbers.getUpdateWindowDims().size() != 0) {
+      return failure();
+    }
+
+    Location loc = op.getLoc();
+
+    // Reshape indices to add the implicit 1x out front.
+    llvm::SmallVector<int64_t> newIndicesShape;
+    llvm::SmallVector<Value> newIndicesDynDims;
+    newIndicesShape.push_back(1);
+    for (auto it : llvm::enumerate(indicesTy.getShape())) {
+      auto dim = it.value();
+      newIndicesShape.push_back(dim);
+    }
+
+    Value reshapedIndices = rewriter.create<mhlo::ReshapeOp>(
+        loc, RankedTensorType::get(newIndicesShape, indicesTy.getElementType()),
+        indices);
+
+    Value reshapedUpdates = rewriter.create<mhlo::ReshapeOp>(
+        loc, RankedTensorType::get({1}, updatesTy.getElementType()), updates);
+
+    auto iota = [](llvm::SmallVector<int64_t> &vec, int start, int end) {
+      for (int i = start; i < end; i++) {
+        vec.push_back(i);
+      }
+    };
+
+    SmallVector<int64_t> updateWindowDims;
+
+    SmallVector<int64_t> insertedWindowDims;
+    iota(insertedWindowDims, 0, operandTy.getRank());
+
+    SmallVector<int64_t> scatterDimsToOperandDims;
+    iota(scatterDimsToOperandDims, 0, operandTy.getRank());
+    auto newDimNumbers = mhlo::ScatterDimensionNumbersAttr::get(
+        op.getContext(), updateWindowDims, insertedWindowDims,
+        scatterDimsToOperandDims, /*indexVectorDim=*/1);
```

Comment:
We don't have to declare `updateWindowDims`. Instead, we can just use `{}` and have a function arg comment. E.g.,

```
mhlo::ScatterDimensionNumbersAttr::get(..., /*updateWindowDims=*/{}, ...);
```